src/pages/version/taskDuration/TaskDurationTable.tsx

```diff
@@ -0,0 +1,259 @@
+            }
+          />,
+        ]}
+      >
+        {({ datum }) => (
+          <DisplayTaskRow
+            task={datum}
+            max={max}
+            data-cy="task-duration-table-row"
+          >
+            {datum?.executionTasksFull &&
+              datum?.executionTasksFull.map((task) => {
+                const {
+                  id,
+                  displayName,
+                  status,
+                  buildVariantDisplayName,
+                  timeTaken,
+                } = task;
+                const barWidth = calculateBarWidth(timeTaken, max);
+                const barColor =
+                  mapTaskToBarchartColor[mapTaskStatusToUmbrellaStatus[status]];
+                return (
+                  <Row key={id}>
+                    <Cell>
+                      <TaskNameWrapper>
+                        {displayName}
+                        <TaskStatusIcon status={status} />
+                      </TaskNameWrapper>
+                    </Cell>
+                    <Cell>{buildVariantDisplayName}</Cell>
+                    <Cell>
+                      <DurationWrapper>
+                        <Bar width={barWidth} color={barColor} />
+                        <TimeLabel>{msToDuration(timeTaken)}</TimeLabel>
+                      </DurationWrapper>
+                    </Cell>
+                  </Row>
+                );
+              })}
+          </DisplayTaskRow>
+        )}
+      </Table>
+      {tasks.length === 0 && <NoTableResults message="No tasks found." />}
+    </TableWrapper>
+  );
+};
+
+interface RowProps {
+  task: PatchTaskDurationsQuery["patchTasks"]["tasks"][0];
+  max: number;
+  "data-cy"?: string;
+  children?: React.ReactNode;
+}
+
+const DisplayTaskRow: React.VFC<RowProps> = ({
+  task,
+  max,
+  children,
+  "data-cy": dataCy,
+}) => {
+  const { id, displayName, status, buildVariantDisplayName, timeTaken } = task;
+  const barWidth = calculateBarWidth(timeTaken, max);
+  const barColor =
+    mapTaskToBarchartColor[mapTaskStatusToUmbrellaStatus[status]];
+  return (
+    <Row key={id} data-cy={dataCy}>
+      <Cell>
+        <TaskNameWrapper>
+          {displayName}
+          <TaskStatusIcon status={status} />
+        </TaskNameWrapper>
+      </Cell>
+      <Cell>{buildVariantDisplayName}</Cell>
+      <Cell>
+        <DurationWrapper>
+          <Bar width={barWidth} color={barColor} />
+          <TimeLabel>{msToDuration(timeTaken)}</TimeLabel>
+        </DurationWrapper>
+      </Cell>
+      {children}
+    </Row>
+  );
+};
+
+const findMax = (tasks: PatchTaskDurationsQuery["patchTasks"]["tasks"]) => {
+  const durations = tasks.map((t) => t.timeTaken);
+  return Math.max(...durations);
+};
+
+const calculateBarWidth = (value: number, max: number) =>
+  max ? `${(value / max) * 100}%` : "0%";
+
+const TableWrapper = styled.div`
+  border-top: 3px solid ${gray.light2};
+  // Styling the Cell directly doesn't work, so styling is done through the table.
```

Comment:
I was able to apply styles directly to the cell.
<img width="433" alt="image" src="https://user-images.githubusercontent.com/4605522/164540913-49eb644e-4a9d-4991-a921-0c29c641cf99.png">

You need to use classNames https://emotion.sh/docs/class-names to style the cells.
I think LG does [some matching under the hood](https://github.com/mongodb/leafygreen-ui/blob/89390382406cf0a45896714f9c8aff17ad35042f/packages/lib/src/index.ts#L61) to render its elements. I think emotion breaks this behavior since it creates a new component with a new display name.
https://github.com/mongodb/leafygreen-ui/blob/main/packages/table/src/Row.tsx

Alternatively you can set the displayName of your styled component to `Cell`  to trick LG into thinking its the same thing.

Reply:
might be worth bring up with them if theres a preferred way.

Reply:
I went with the second option (setting the displayName), but only because using the `ClassNames` notation breaks the execution tasks 🥲

I think Leafygreen's nested row policy must be strict, it seems like `Row` needs to be the direct descendant in order for it to work...